src/zenml/cli/artifact.py

```diff
@@ -83,37 +90,76 @@ def list_artifacts(**kwargs: Any) -> None:
     only_metadata: bool = False,
     yes: bool = False,
 ) -> None:
-    """Delete an artifact.
+    """Delete an artifact by ID or name.
+
+    If an artifact name without a version is provided, all artifacts with that
+    name will be deleted.
 
     Args:
-        artifact_id: ID of the artifact to delete.
+        artifact_name_or_id: Name or ID of the artifact to delete.
+        version: Version of the artifact to delete.
         only_artifact: If set, only delete the artifact but not its metadata.
         only_metadata: If set, only delete metadata and not the actual artifact.
         yes: If set, don't ask for confirmation.
     """
+    is_uuid = is_valid_uuid(artifact_name_or_id)
+
     if not yes:
-        confirmation = cli_utils.confirmation(
-            f"Are you sure you want to delete artifact '{artifact_id}'?"
-        )
+        if is_uuid:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete artifact "
+                f"'{artifact_name_or_id}'?"
+            )
+        elif version:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete version '{version}' of "
+                f"artifact '{artifact_name_or_id}'?"
+            )
+        else:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete all artifacts with name "
+                f"'{artifact_name_or_id}'?"
+            )
         if not confirmation:
             cli_utils.declare("Artifact deletion canceled.")
             return
 
     try:
-        Client().delete_artifact(
-            artifact_id=UUID(artifact_id),
-            delete_metadata=not only_artifact,
-            delete_from_artifact_store=not only_metadata,
-        )
+        versions: List[Optional[str]] = [version]
+        if not is_uuid and not version:
+            artifact_models = depaginate(
+                partial(Client().list_artifacts, name=artifact_name_or_id)
+            )
+            versions = [
+                artifact_model.version for artifact_model in artifact_models
+            ]
+        for version_ in versions:
+            Client().delete_artifact(
+                name_id_or_prefix=artifact_name_or_id,
+                version=version_,
+                delete_metadata=not only_artifact,
+                delete_from_artifact_store=not only_metadata,
+            )
     except (KeyError, ValueError) as e:
         cli_utils.error(str(e))
     else:
-        cli_utils.declare(f"Artifact '{artifact_id}' deleted.")
+        if is_uuid:
+            cli_utils.declare(f"Artifact '{artifact_name_or_id}' deleted.")
+        elif version:
+            cli_utils.declare(
+                f"Version '{version}' of artifact '{artifact_name_or_id}' "
+                f"deleted."
+            )
+        else:
+            cli_utils.declare(
+                f"All artifacts with name '{artifact_name_or_id}' deleted."
+            )
```

Comment:
Same question here - why is this different from uuid case?

Reply:
each artifact version has a different ID, so if you specify an ID you only delete that version, whereas for a name there are multiple versions associated to it

Reply:
Ah, gotca